plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -623,16 +623,16 @@ protected boolean isSupportedJoinCondition(JdbcJoinCondition joinCondition)
         }
 
         String typeName = typeHandle.getJdbcTypeName().get();
-        if (typeName.equalsIgnoreCase("tinyint unsigned")) {
+        if (typeName.equalsIgnoreCase("tinyint(3) unsigned")) {
```

Comment:
Is it instead possible to create the proper JdbcTypeHandle for such columns in `JdbcClient#getColumns`? See for example how Postgres overrides that to fill in information about array types.

This seems prone to breakage and I'm not sure if this depends on the exact server and driver versions being used.

Reply:
If this isn't possible to do then I suggest we drop this migration for now.

I'm of the opinion that if switching the driver isn't bringing any benefits at the moment then we should stick with the mariadb driver for now.
I looked at the [current diff](https://github.com/mariadb-corporation/mariadb-connector-j/compare/master...memsql:master) and don't see any improvements to consider the switch.

Reply:
@hashhar - I had a questions with @wendigo about whether this list is exhaustive or not (I look in their source and I see tinyint(1)->tinyint(4), so I expect it is not). I agree it's fragile and could fail unexpectedly unless we had coverage of all possible types.
 I'll look at the JdbcTypeHandle/Postgress client.getColumns - thanks for the suggestion!
(Whether we postpone the change or not, at least this is a useful learning experience for me! :+1: )

Reply:
Considering the issues with the change to the connection string (thanks @jhlodin for raising that), it looks like a bigger change than originally envisioned, and for very little benefit (as you say, their current driver doesn't add much, and complicates the column  mapping - future versions may change further, etc).

Reply:
Maybe we can revisit this PR in the future once their driver matures a bit. WDYT @wendigo ?

Reply:
Update / clarification: I'm going to attempt to update the type mapping as suggested, along with making `jdbc:mariadb` urls work transparently (that's already done w/tests, etc).
I may close this PR + re-open a new one though - so shout if I should do that! 👍